### DISCUSSION TITLE
YAML based configuration management for GitHub settings?

### DISCUSSION THREAD

**Discussion by ssbarnea** [community/community#40545] [Apps, API and Webhooks]
I am looking for a tool to manage lots of GitHub settings including:
- repositories
- labels
- organizations
- teams

Can someone recommend such a solution? I am interested in particular on one using YAML as that makes it very easy to managed while also allowing avoiding repetitions (anchors).

It needs to scale as one of the cases I am looking for is to manage >200 repos on 5 organizations, so it needs to be quite smart about how to use github API due to rate limits.

**Comment by tuves** [comment]
Welcome to the GitHub Community, @ssbarnea, we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Reply by ssbarnea** [reply]
Only two categories looked as suitable for this question: General and Discover, I picked "Discover". Now, I do not know why I see in API and Webhooks,... maybe I pressed the wrong button or someone else moved. Anyway, I do do not think that is wrong either.

**Comment by ssbarnea** [comment]
I was able to find https://github.com/googleapis/github-repo-automation but I did not test it, so anyone is welcomed to provide feedback on it or suggest another one.

**Comment by merusso** [comment]
You can manage GitHub settings with Terraform using Terraform's [GitHub provider](https://registry.terraform.io/providers/integrations/github/latest).
